File: ゼミ用/05製造/api/src/request/CreateAccountRequest.php
<?php

/**
 * アカウント登録処理の値を保持する
 */
class CreateAccountRequest {
    private $userName;
    private $password;
    private $errorMsg = array();

    /**
     * アカウント登録処理の値を格納します
     */
    public function __construct($userName, $password) {
        $this->userName = $userName;
        $this->password = $password;
    }

    public function getUserName() {
        return $this->userName;
    }

    public function getPassword() {
        return $this->password;
    }

    public function getErrorMsg() {
        return $this->errorMsg;
    }

    /**
     * バリデーションチェックをします
     */
    public function validation() {
        $validationFlg = false;

        // ユーザ名の文字数チェック
        if (20 < strlen($this->userName)) {
            array_push($this->errorMsg, 'エラー ：ユーザ名は20文字以内で入力してください');
            $validationFlg = true;
        }

        // パスワードの文字数チェック
        if (20 < strlen($this->password)) {
            array_push($this->errorMsg, 'エラー ：パスワードは20文字以内で入力してください');
            $validationFlg = true;
        }

        // 入力項目チェック
        if (empty($this->userName) || empty($this->password)) {
            array_push($this->errorMsg, 'エラー ：必須項目が入力されていません');
            $validationFlg = true;
        }

        return $validationFlg;
    }
}

File: ゼミ用/05製造/eeldev/eelPy/api/request/CreateAccountRequest.py
# /**
# * アカウント登録処理の値を保持する
# */
class CreateAccountRequest:
    errorMsg = []

    # /**
    # * アカウント登録処理の値を格納します
    # */
    def __init__(self, userName, password):
        self.userName = userName
        self.password = password

    def getUserName(self):
        return self.userName

    def getPassword(self):
        return self.password

    def getErrorMsg(self):
        return self.errorMsg

    # /**
    # * バリデーションチェックをします
    # */
    def validation(self):
        validationFlg = False

        try:
            # 入力項目チェック
            if (len(self.userName) == 0 or len(self.password) == 0):
                self.errorMsg.append('エラー ：必須項目が入力されていません')
                validationFlg = True

            # ユーザ名の文字数チェック
            if (20 < len(self.userName)):
                self.errorMsg.append('エラー ：ユーザ名は20文字以内で入力してください')
                validationFlg = True

            # パスワードの文字数チェック
            if (20 < len(self.password)):
                self.errorMsg.append('エラー ：パスワードは20文字以内で入力してください')
                validationFlg = True

        except Exception as e:
            validationFlg = True

        return validationFlg
